platform/core/src/runs/TransactingMutationListener.js

```diff
@@ -38,9 +44,28 @@ define([], function () {
                     transactionService.startTransaction();
                 }
 
+                if (!MUTATION_TRACKER.has(domainObject)) {
+                    MUTATION_TRACKER.set(domainObject, domainObject
+                        .getCapability('mutation')
+                        .listen(function () {})
+                    );
+                }
+
+                cacheService.put(domainObject.getId(), domainObject.getModel());
+
+                function unlistenAndCall(f) {
+                    return function () {
+                        if (MUTATION_TRACKER.has(domainObject)) {
```

Comment:
@VWoeltjen this guarantees only one listener per domain object instance, while mutation can be called on the same object instance multiple times.  This will only unlisten a single time (it removes it from the tracker afterwards).